source/state_representation/README.md

```diff
@@ -1,146 +1,622 @@
+    * [Reference frame](#reference-frames)
+    * [Construction](#construction)
+    * [Cartesian getters and setters](#cartesian-getters-and-setters)
+    * [Cartesian addition and subtraction](#cartesian-addition-and-subtraction)
+    * [Cartesian transforms: changing the reference frame](#cartesian-transforms--changing-the-reference-frame)
+* [Derived Cartesian classes](#derived-cartesian-classes)
+    * [Cartesian pose](#cartesian-pose)
+    * [Cartesian twist](#cartesian-twist)
+    * [Cartesian acceleration](#cartesian-acceleration)
+    * [Cartesian wrench](#cartesian-wrench)
+    * [Cartesian integration and derivation](#cartesian-integration-and-derivation)
 * [Joint state](#joint-state)
-  * [Joint state operations](#joint-state-operations)
-  * [Conversion between joint state variables](#conversion-between-joint-state-variables)
+    * [Joint state operations](#joint-state-operations)
+    * [Conversion between joint state variables](#conversion-between-joint-state-variables)
 * [The Jacobian matrix](#the-jacobian-matrix)
-  * [Conversion between JointVelocities and CartesianTwist](#conversion-between-jointvelocities-and-cartesiantwist)
-  * [Conversion between JointTorques and CartesianWrench](#conversion-between-jointtorques-and-cartesianwrench)
-  * [Matrix multiplication](#matrix-multiplication)
-  * [Changing the Jacobian reference frame](#changing-the-jacobian-reference-frame)
+    * [Conversion between JointVelocities and CartesianTwist](#conversion-between-jointvelocities-and-cartesiantwist)
+    * [Conversion between JointTorques and CartesianWrench](#conversion-between-jointtorques-and-cartesianwrench)
+    * [Matrix multiplication](#matrix-multiplication)
+    * [Changing the Jacobian reference frame](#changing-the-jacobian-reference-frame)
+
+## State
+
+The `State` base class defines the following attributes common to all derived states:
+
+- name: the name associated with the state
+- type: the specific StateType of the object
+- timestamp: an internal timestamp which refers to the last modification
+
+### Name
+
+The name is used to label a specific state instance, and can be used to disambiguate multiple states or check their
+compatibility. The name can be accessed or modified with `get_name()` and `set_name()` respectively.
+
+### State Type
+
+A state can hold different data depending on its type. The available state types are defined by the
+`state_representation::StateType` enumeration. The type field is accessible using the `get_type()` method and allows
+introspection when working with state pointers. It is a readonly property, determined by the constructor implementation.
+
+```c++
+void check_state_type(const std::shared_ptr<state_representation::State>& state) {
+  if (state->get_type() == ...) {
+    // 
+  }
+}
+```
+
+### Timestamp
+
+The timestamp of a state records when it was last modified. Any non-const operations or mutations,
```

Comment:
My opinion remains the same, so setting the name should also reset the timestamp. This will be down to the implementation of #15 before release but I would rather follow this strict and explicit ruling consistently.